### PR TITLE
Add calendar versioning

### PR DESCRIPTION
**Pull request by CaseyHillers**

### CODE REVIEW

File: lib/src/common/github.dart [hunk 1]
```diff
@@ -29,12 +30,24 @@ class GitHub {
...
   final String endpoint;
 
+  /// Calendar version of the GitHub API to use.
+  ///
+  /// Changing this value is unsupported. However, it may unblock you if there's
+  /// hotfix versions.
+  ///
+  /// See also:
+  ///   * https://docs.github.com/en/rest/overview/api-versions?apiVersion=2022-11-28
+  final String version;
```
Comment (CaseyHillers) [inline lib/src/common/github.dart:49]: Most of this change was me debating if we should make this private. I lean towards making it public to give an easy escape hatch for downstream dependencies to override what the library is using.

**Review by kevmoo — APPROVED** [review]
Do we want a changelog entry here?

**Comment by CaseyHillers** [comment]
> Do we want a changelog entry here?

Based on https://semver.org/#spec-item-7, that's expected. We're adding `version` to the main entry point of this library.

I can't recall if the release automation generates the changelog entries, so I opt'd to add it here.

**Comment by kevmoo** [comment]
> > Do we want a changelog entry here?
> 
> Based on https://semver.org/#spec-item-7, that's expected. We're adding `version` to the main entry point of this library.
> 
> I can't recall if the release automation generates the changelog entries, so I opt'd to add it here.

Auto-magic drives me nuts. Okay.

**Comment by robrbecker** [comment]
Yep, The changelog and version will be updated automatically after this merges.

**Review by robrbecker — APPROVED** [review]

**Comment by robrbecker** [comment]
Released in version 9.7.0 https://github.com/SpinlockLabs/github.dart/releases/tag/9.7.0

**Pull request merged**